src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -187,6 +176,33 @@ public TaskResult call() {
         return new TaskResult(exception);
     }
 
+    private void createLeasesForChildShardsIfNotExist() throws InvalidStateException, DependencyException, ProvisionedThroughputException {
+        for (ChildShard childShard : childShards) {
+            final String leaseKey = childShard.getShardId();
+            if (leaseCoordinator.getCurrentlyHeldLease(leaseKey) == null) {
+                final KinesisClientLease leaseToCreate = KinesisShardSyncer.newKCLLeaseForChildShard(childShard);
+                leaseCoordinator.getLeaseManager().createLeaseIfNotExists(leaseToCreate);
+                LOG.info("Shard " + shardInfo.getShardId() + " : Created child shard lease: " + leaseToCreate.getLeaseKey());
+            }
+        }
+    }
+
+    private void updateCurrentLeaseWithChildShards() throws DependencyException, InvalidStateException, ProvisionedThroughputException {
+        final KinesisClientLease currentLease = leaseCoordinator.getCurrentlyHeldLease(shardInfo.getShardId());
```

Comment:
Should this be using #getLease like above? I believe it will work as is, but the "Failed to retrieve" message might be confusing if we're not actually retrieving from DDB

Reply:
If we are getting the lease for the current shard, it's guaranteed that the shard is in local cache because current worker must be holding the lease. We want to avoid unnecessary transaction with DDB if the information can be got from cache